src/main/java/com/cloudbees/jenkins/GitHubPushCause.java

```diff
@@ -37,5 +37,17 @@ public GitHubPushCause(File pollingLog, String pusher) throws IOException {
     public String getShortDescription() {
         return format("Started by GitHub push by %s", trimToEmpty(pushedBy));
     }
+
+    @Override
+    public boolean equals(Object o) {
+        return o instanceof GitHubPushCause || super.equals(o);
```

Comment:
This is a bug IIUC. This must be an `&&`, or any `GitHubPushCause` instance will be seen as equal?

Reply:
Good catch, I also need to compare `pushedBy`

Reply:
Does this look correct?

```
    @Override
    public boolean equals(Object o) {
        return o instanceof GitHubPushCause &&
                (null != o && this.pushedBy == ((GitHubPushCause) o).pushedBy) &&
                super.equals(o);
    }
```

Reply:
This should be resolved.